File: src/main/java/fi/metatavu/metaform/server/rest/ReplyMode.java
package fi.metatavu.metaform.server.rest;

/**
 * Enumeration that decribes reply mode
 * 
 * @author Antti Leppä
 * @author Heikki Kurhinen
 */
public enum ReplyMode {
  
  /**
   * Updates existing reply if one exists, otherwise create new
   */
  UPDATE,
  
  /**
   * Revision existing reply and create new one
   */
  REVISION,
  
  /**
   * Always create new reply 
   */
  CUMULATIVE

}

File: src/main/kotlin/fi/metatavu/metaform/server/rest/ReplyMode.kt
package fi.metatavu.metaform.server.rest

enum class ReplyMode {

    /**
     * Updates existing reply if one exists, otherwise create new
     */
    UPDATE,

    /**
     * Revision existing reply and create new one
     */
    REVISION,

    /**
     * Always create new reply
     */
    CUMULATIVE
}